src/EmployerFeedback/SFA.DAS.EmployerFeedback/Application/Commands/SyncEmployerAccounts/SyncEmployerAccountsCommandHandler.cs

```diff
@@ -0,0 +1,131 @@
+using MediatR;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.EmployerFeedback.InnerApi.Requests;
+using SFA.DAS.EmployerFeedback.InnerApi.Responses;
+using SFA.DAS.SharedOuterApi.Configuration;
+using SFA.DAS.SharedOuterApi.Extensions;
+using SFA.DAS.SharedOuterApi.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.EmployerFeedback.Application.Commands.SyncEmployerAccounts
+{
+    public class SyncEmployerAccountsCommandHandler : IRequestHandler<SyncEmployerAccountsCommand>
+    {
+        private readonly IEmployerFeedbackApiClient<EmployerFeedbackApiConfiguration> _feedbackApiClient;
+        private readonly IAccountsApiClient<AccountsConfiguration> _accountsApiClient;
+        private readonly ILogger<SyncEmployerAccountsCommandHandler> _logger;
+        private const int MaxRetryAttempts = 3;
+        private const int PageSize = 1000;
+
+        public SyncEmployerAccountsCommandHandler(
+            IEmployerFeedbackApiClient<EmployerFeedbackApiConfiguration> feedbackApiClient,
+            IAccountsApiClient<AccountsConfiguration> accountsApiClient,
+            ILogger<SyncEmployerAccountsCommandHandler> logger)
+        {
+            _feedbackApiClient = feedbackApiClient;
+            _accountsApiClient = accountsApiClient;
+            _logger = logger;
+        }
+
+        public async Task Handle(SyncEmployerAccountsCommand request, CancellationToken cancellationToken)
+        {
+            DateTime syncStartTime = DateTime.UtcNow;
+            DateTime? lastSyncDate = await GetLastSyncDateAsync(cancellationToken);
+
+            int page = 1;
+            int totalPages;
+            do
+            {
+                var accountsResponse = await GetAccountsPageAsync(lastSyncDate, page, cancellationToken);
+                if (accountsResponse == null)
+                {
+                    _logger.LogInformation(
+                        "SyncEmployerAccounts: Received null response when fetching page {Page}. LastSyncDate={LastSyncDate}, SyncStartTime={SyncStartTime}",
+                        page, lastSyncDate, syncStartTime);
+                    break;
```

Comment:
When this errors - it still updates the sync time, so the next run will ignore those changes - that is not correct